lib/Attachment.php

```diff
@@ -156,6 +156,53 @@ public function __toString() {
 	 * @param int|mixed $iid An attachment identifier.
 	 */
 	public function init( $iid = null ) {
+		$iid = $this->determine_id( $iid );
+
+		/**
+		 * determine_id returns null when the attachment is a file path,
+		 * thus there's nothing in the DB for us to do here 
+		 */
+		if ( null === $iid ) {
+			return null;
+		}
+
+		$attachment_info = $this->get_attachment_info( $iid );
```

Comment:
Would it be clearer if we moved this part to a separate method, `init_from_attachment()` maybe?

Reply:
I'd prefer to keep this section in the main `init` section. All `inits` are not created equal and the "regular" course of action for an Attachment should be `init`ing as an attachment. It's the other cases: when the data is a URL, file path, etc., that are "special" and deserve to be processed elsewhere